pyccel/codegen/printing/ccode.py

```diff
@@ -1106,8 +1106,10 @@ def _print_IndexedElement(self, expr):
                     if isinstance(ind, Slice):
                         inds[i] = self._new_slice_with_processed_arguments(ind, PyccelArraySize(base, i),
                             allow_negative_indexes)
+                        inds[i].slice_type = Slice.Range
                     else:
                         inds[i] = Slice(ind, PyccelAdd(ind, LiteralInteger(1), simplify = True), LiteralInteger(1))
+                        inds[i].slice_type = Slice.Element
```

Comment:
Shouldn't the constructor argument be sufficient to fix the `slice_type`?

Reply:
https://github.com/pyccel/pyccel/pull/1299/commits/6331896f3ba794243b8f89d73913440e41f7b0f2